src/python/zquantum/core/cost_function.py

```diff
@@ -83,14 +86,15 @@ def get_gradient(self, parameters: np.ndarray) -> np.ndarray:
             raise Exception("Gradient type: %s is not supported", self.gradient_type)
 
 
-class EvaluateOperatorCostFunction(CostFunction):
+class AnsatzBasedCostFunction(CostFunction):
```

Comment:
A think tests should be updated appropriately.

Reply:
Here and elsewhere as well.

Reply:
Will do. Thanks for taking the first pass, I'll do tests tomorrow 😄